### PR TITLE
D3ASIM-2948: Fixed the one sided market trade price to take into acco…

### PR DESCRIPTION
**Pull request by spyrostz**
…unt grid fees.

### CODE REVIEW

File: src/d3a/models/strategy/area_agents/one_sided_engine.py [hunk 1]
```diff
@@ -143,7 +143,13 @@ def event_trade(self, *, trade):
...
 
             try:
-                trade_offer_rate = trade.offer.energy_rate
+                if ConstSettings.IAASettings.MARKET_TYPE == 1 and trade is not None:
+                    # One sided market should subtract the fees
+                    trade_offer_rate = trade.offer.energy_rate - \
+                                       trade.fee_price / trade.offer.energy
+                else:
+                    # trade_offer_rate not used in two sided markets, trade_bid_info used instead
+                    trade_offer_rate = None
```
Comment (hannesdiedrich) [inline src/d3a/models/strategy/area_agents/one_sided_engine.py:152]: really? not `trade_offer_rate = trade.offer.energy_rate` ?

Comment (hannesdiedrich) [reply src/d3a/models/strategy/area_agents/one_sided_engine.py:152]: Ah yes, in the accept_offer, the offer rate is used then.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

File: src/d3a/models/strategy/area_agents/one_sided_engine.py (hunk 1 above)
Comment (spyrostz) [reply src/d3a/models/strategy/area_agents/one_sided_engine.py:152]: It is not used at all in the 2-sided market indeed. The clause you mention is already handled by the market internally https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/market/one_sided.py#L214

**Comment by hannesdiedrich** [comment]
Thanks a lot for dealing with this so quickly. I wonder whether https://github.com/gridsingularity/d3a/pull/914 should be reverted now. (also maybe the market dependant behaviour in the bills?

File: src/d3a/models/strategy/area_agents/one_sided_engine.py [hunk 2]
```diff
@@ -143,7 +143,13 @@ def event_trade(self, *, trade):
                 f"offer: source_rate ({source_rate}) is not lower than target_rate ({target_rate})"
 
             try:
-                trade_offer_rate = trade.offer.energy_rate
+                if ConstSettings.IAASettings.MARKET_TYPE == 1 and trade is not None:
```
Comment (faizan2590) [inline src/d3a/models/strategy/area_agents/one_sided_engine.py:146]: I think the second check `..and trade is not None` could be avoided as at its been validated at R138 at this point of commit ?

Comment (spyrostz) [reply src/d3a/models/strategy/area_agents/one_sided_engine.py:146]: Good catch. Fixing it, thanks!

**Comment by spyrostz** [comment]
> Thanks a lot for dealing with this so quickly. I wonder whether #914 should be reverted now. (also maybe the market dependant behaviour in the bills?

Thank you, this actually exhibited in a test failure! Fixing this

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**